src/bin/pgcopydb/cli_snapshot.c

```diff
@@ -290,6 +308,23 @@ cli_create_snapshot(int argc, char **argv)
 
 		bool logSQL = log_get_level() <= LOG_TRACE;
 
+		/*
+		 * When using pgcopydb snapshot --follow --restart we first cleanup the
+		 * previous setup, and that includes dropping the replication slot.
+		 */
+		if (createSNoptions.restart)
+		{
+			log_info("Clean-up replication setup, per --restart");
+
+			if (!stream_cleanup_databases(&copySpecs,
```

Comment:
@dimitri I see this function call also drops replication origin on the target., the snapshot never dealt with the target so far right? I'm not sure we should clean up the origin & sentinel tables as part of this.? 

I think this option adds a bit of confusion around the responsibilities of the subcommand(snapshot, stream cleanup).